SingularityService/src/main/java/com/hubspot/singularity/smtp/MailTemplateHelpers.java

```diff
@@ -0,0 +1,216 @@
+package com.hubspot.singularity.smtp;
+
+import com.google.common.base.Optional;
+import com.google.common.collect.ImmutableMap;
+import com.google.common.collect.Lists;
+import com.hubspot.mesos.MesosUtils;
+import com.hubspot.mesos.json.MesosFileChunkObject;
+import com.hubspot.singularity.ExtendedTaskState;
+import com.hubspot.singularity.SingularityTask;
+import com.hubspot.singularity.SingularityTaskHistoryUpdate;
+import com.hubspot.singularity.SingularityTaskId;
+import com.hubspot.singularity.config.EmailConfigurationEnums;
+import com.hubspot.singularity.config.SMTPConfiguration;
+import com.hubspot.singularity.data.SandboxManager;
+import org.apache.commons.lang3.text.WordUtils;
+import org.apache.commons.lang3.time.DateFormatUtils;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import java.util.Collection;
+import java.util.List;
+import java.util.Map;
+
+/**
+ * Helpers that get passed to the Jade renderer. These helpers manipulate information given to the
+ * Jade context into different formats.
+ */
+public class MailTemplateHelpers {
+  private static final Logger LOG = LoggerFactory.getLogger(SingularityMailer.class);
+
+  /**
+   * For fetching log files from mesos slaves.
+   */
+  private final SandboxManager sandboxManager;
+
+  private static final String TASK_DATE_PATTERN = "MMM dd HH:mm:ss";
+  private static final String TASK_LINK_FORMAT = "%s/task/%s";
+  private static final String REQUEST_LINK_FORMAT = "%s/request/%s";
+  private static final String LOG_LINK_FORMAT = "%s/task/%s/tail/%s";
+
+  /**
+   * Used to generate links, if no String is present, helpers will return empty Strings.
+   */
+  private final Optional<String> uiHostnameAndPath;
+  private final Optional<SMTPConfiguration> maybeSmtpConfiguration;
+
+  public MailTemplateHelpers(Optional<String> uiHostnameAndPath, SandboxManager sandboxManager, Optional<SMTPConfiguration> maybeSmtpConfiguration) {
+    this.uiHostnameAndPath = uiHostnameAndPath;
+    this.sandboxManager = sandboxManager;
+    this.maybeSmtpConfiguration = maybeSmtpConfiguration;
+  }
+
+  /**
+   * Format taskHistory information into a Map for Jade to generate a table from.
+   * @param taskHistory task history information.
+   * @return map for Jade to pull "date", "update", and "message" information from.
+   */
+  public List<Map<String, String>> getJadeTaskHistory(Collection<SingularityTaskHistoryUpdate> taskHistory) {
+    List<Map<String, String>> output = Lists.newArrayListWithCapacity(taskHistory.size());
+
+    for (SingularityTaskHistoryUpdate taskUpdate : taskHistory) {
+      output.add(
+          ImmutableMap.<String, String> builder()
+              .put("date", DateFormatUtils.formatUTC(taskUpdate.getTimestamp(), TASK_DATE_PATTERN))
+              .put("update", WordUtils.capitalize(taskUpdate.getTaskState().getDisplayName()))
+              .put("message", taskUpdate.getStatusMessage().or(""))
+              .build());
```

Comment:
instead of a Map<>, this should probably be a `SingularityMailTaskHistoryUpdate` class